src/app/modules/auth.rs

```diff
@@ -1,18 +1,246 @@
-use crate::app::store::ProvableStore;
-use crate::app::BaseCoinApp;
+use crate::app::modules::bank::Denom;
+use crate::app::modules::Module;
+use crate::app::store::{
+    Height, Path, ProtobufStore, ProvableStore, SharedStore, Store, TypedStore,
+};
+use crate::prostgen::cosmos::auth::v1beta1::BaseAccount;
 use crate::prostgen::cosmos::auth::v1beta1::{
-    query_server::Query as AuthQuery, QueryAccountRequest, QueryAccountResponse,
-    QueryAccountsRequest, QueryAccountsResponse, QueryParamsRequest as AuthQueryParamsRequest,
-    QueryParamsResponse as AuthQueryParamsResponse,
+    query_server::{Query, QueryServer},
+    QueryAccountRequest, QueryAccountResponse, QueryAccountsRequest, QueryAccountsResponse,
+    QueryParamsRequest, QueryParamsResponse,
 };
 
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::str::FromStr;
+
+use cosmrs::AccountId;
 use prost::Message;
 use prost_types::Any;
+use serde_json::Value;
+use tendermint_proto::Protobuf;
 use tonic::{Request, Response, Status};
-use tracing::debug;
+use tracing::{debug, trace};
+
+const RELAYER_ACCOUNT: &str = "cosmos1snd5m4h0wt5ur55d47vpxla389r2xkf8dl6g9w";
+
+#[derive(Clone)]
+struct AccountsPath(AccountId);
+
+impl From<AccountsPath> for Path {
+    fn from(path: AccountsPath) -> Self {
+        format!("accounts/{}", path.0).try_into().unwrap() // safety - cannot fail as AccountsPath is correct-by-construction
+    }
+}
+
+pub trait Account {
+    type Address;
+    type PubKey;
+
+    fn address(&self) -> &Self::Address;
+    fn pub_key(&self) -> &Self::PubKey;
+    fn number(&self) -> u64;
+    fn sequence(&self) -> u64;
```

Comment:
Can we document these methods? Particularly it's not clear to me what `number()` and `sequence() are

Reply:
As discussed these methods were taken from Cosmos-SDK's `AuthKeeper`, but let's remove the `sequence()` as basecoin doesn't support pruning. I will document these methods.

Reply:
https://github.com/informalsystems/basecoin-rs/pull/44/commits/75b4063ee6c6540face444b54e29ca27d0a39447